### PR TITLE
Adds some hyperlink functionality

### PR DESCRIPTION
**Pull request by the-og-gear**
Used some regex and JS .replace function. Enjoy.

### CODE REVIEW

**Comment by the-og-gear** [comment]
image
Collects links as expected, and gives a valid tag for when you click. Only thing I haven't done is modify any CSS because honestly I'll just make it look horrible.

File: code/modules/vchat/js/vchat.js [hunk 1]
```diff
@@ -578,6 +578,10 @@ function start_vue() {
 					}
 				}
 
+				newmessage.content = newmessage.content.replace(
+					/(\b(https?|ftp|file):\/\/[\-A-Z0-9+&@#\/%?=~_|!:,.;]*[\-A-Z09+&@#\/%=~_|])/img, //Honestly good luck with this regex ~Gear
```
Comment (Arokha) [inline code/modules/vchat/js/vchat.js:582]: Seems good but allowing `file://` links seems inappropriate given that people could link people to .bat or .exe files on their own computer or do other silly things that might be undesirable in the very old version of IE that byond is using. Maybe just `https?` given that ftp is silly.

Comment (the-og-gear) [reply code/modules/vchat/js/vchat.js]: Oop, fixed.

**Comment by Atermonera** [comment]
absolute regex

**Pull request merged**